File: server-java/src/main/java/com/jixiao2/server/evaluation/AwardTypeController.java
package com.jixiao2.server.evaluation;

import com.jixiao2.server.web.CurrentUser;
import java.util.Map;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/admin/award-types")
public class AwardTypeController {

  private final EvaluationService evaluationService;

  public AwardTypeController(EvaluationService evaluationService) {
    this.evaluationService = evaluationService;
  }

  @GetMapping
  public Map<String, Object> list(@CurrentUser String userId) {
    return evaluationService.listAwardTypes(userId);
  }
}

File: server/modules/evaluation/award-type.controller.ts
import { Controller, Get, Req, UseGuards } from '@nestjs/common';
import type { Request } from 'express';
import { SessionUserGuard } from '../../guards/session-user.guard';
import { EvaluationService } from './evaluation.service';

@Controller('api/admin/award-types')
@UseGuards(SessionUserGuard)
export class AwardTypeController {
  constructor(private readonly evaluationService: EvaluationService) {}

  @Get()
  list(@Req() req: Request) {
    return this.evaluationService.listAwardTypes(req.userContext!.userId!);
  }
}
